packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -535,5 +554,20 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
      * Gets a form of the object that can be serialized.
      * @returns A tree representing the snapshot of the shared object.
      */
-    protected abstract summarizeCore(serializer: IFluidSerializer): ISummaryTreeWithStats;
+    protected abstract summarizeCore(
+        serializer: IFluidSerializer,
+        telemetryContext?: ITelemetryContext,
+    ): ISummaryTreeWithStats;
+
+    protected incrementSummarizeInstanceCount(telemetryContext?: ITelemetryContext): void {
+        const instanceCountProperty = "InstanceCount";
+        let instanceCount = telemetryContext?.get(this.telemetryContextPrefix, instanceCountProperty) ?? 0;
+        telemetryContext?.set(this.telemetryContextPrefix, instanceCountProperty, ++instanceCount);
```

Comment:
Consider having incr() method in addition to set().
Or add() - a bit more versatile.